### DISCUSSION TITLE
is it possible to use hx-trigger with JS API - htmx.ajax() ?

### DISCUSSION THREAD

**Discussion by BobSquarePants** [bigskysoftware/htmx#1508] [Q&A]
Hi everyone,

is it possible to limit the rate (trough throttle or delay) like with[ hx-trigger](https://htmx.org/attributes/hx-trigger/) but with the JS API [htmx.ajax()](https://htmx.org/api/#ajax) ?

I've tried something like
```python
window.htmx.ajax('POST', '/somewhere', {'target':'#here', 'swap':'innerHTML','event':'click throttle:2s'})
```
so it's work, but without any restrictions on the throttling. 
(I guess that the name for JS API for trigger is event ? I see no reference of trigger for the JS API ) 

Thanks.

**Comment by andryyy** [comment]
There would need to be a way to call getTriggerSpecs() from context data, I guess.

Could be added to an event object or a new parameter like "specs". :)

**Reply by andryyy** [reply]
I wonder if you could omit the event in the ajax request and add a hx-trigger attribute to the target element.

**Reply by andryyy** [reply]
Or, if an event is defined, htmx may dynamically set the event value as string to the target before processing. This assumes that the ajaxHelper is capable to process this. I’m on mobile right now and cannot really check this.
